9.0 account fiscal year

damdam-s opened the pull request.

damdam-s committed:
[ADD] add module "account_fiscal_year"

damdam-s committed:
[IMP] add a method on date_range object to retrieve FY start date from a date

damdam-s committed:
[IMP] add details on summary

damdam-s committed:
[ADD] readme.txt on modules

eLBati committed:
FIX

  File "/home/elbati/workspace/odoo/instances/demo9-community/parts/odoo-ocb/openerp/sql_db.py", line 220, in execute
    res = self._obj.execute(query, params)
DataError: value too long for type character varying(256)

lmignon committed:
[FIX] Tests

damdam-s committed:
[FIX] try to find a FY start date according to the start date from
choosen period
[IMP] date fields required in ledger wizard

damdam-s committed:
[IMP] get module account_fiscal_year from
oca-sorrento/account-financial-reporting::push-here

sbidoul commented on account_fiscal_year/models/date_range.py:
```diff
@@ -0,0 +1,25 @@
+class DateRange(models.Model):
+    _inherit = 'date.range'
+
+    def find_daterange_fy_start(self, date):
```
This method seems to be unused.

damdam-s replied:
@sbidoul used in another module there: https://github.com/oca-sorrento/account-financial-reporting/blob/push-here/account_financial_report_qweb/wizard/ledger_report_wizard.py#L150

sbidoul replied:
Ok, thanks. I'm not sure why a different method is needed in addition to find_daterange_fy on res.company, though.

damdam-s replied:
Oh ok. I see what you meant. I remove it and update the code in the module where this method was called.

sbidoul commented:
I suggest making the fiscal_year flag read-only, and preventing the unlink of date range of type fiscal year.

Can you also add Laurent and Lorenzo to the contributors?

Plus some polishing of the readme, and it's good to me.

sbidoul commented:
Perhaps it's also this module that should add menus item to access date range from the accounting section.

damdam-s commented:
@sbidoul I will do the points you mentionned before

damdam-s committed:
[IMP] add contributors to README
[IMP] unable to unlink a date_range with type fiscal_year
[IMP] `fiscal_year` flag readonly
[IMP] add menu to date_range under accounting section

damdam-s committed:
[FIX] travis and oca_dependencies files

damdam-s commented:
@sbidoul fix done. Issue with runbot dependency on `server-tools` repository which is probably missing. Can you add it on runbot ?

sbidoul commented:
@damdam-s runbot updated and build restarted.

damdam-s committed:
[FIX] remove method on  object because it's the same as in  file

sbidoul commented on account_fiscal_year/models/date_range.py:
```diff
@@ -0,0 +1,22 @@
+    def unlink(self):
+        """
+        Cannot delete a date_range of type 'fiscal_year'
+        """
```
I'd rather prevent deletion of `date.range.type`, what do you think?

damdam-s replied:
Yes. I think both can be useful. One for not being able to delete `date.range.type` and another one for not being able to delete a `date.range` that use a `date.range.type` of type `fiscal_year`.
I will add the one on `date.range.type`.
Should I keep the other one on `date.range` ?

sbidoul replied:
I would _not_ forbid deletion of date.range. Should that become necessary on some specific contexts it can be done with a record rule.

On the contrary, preventing deletion of date.range.type is necessary for preserving correct behaviour of the module.

damdam-s replied:
Ok I understand. Done in the next commit

damdam-s committed:
[FIX] unable to delete  with flag 'fiscal_year' but can delete

sbidoul commented:
Thanks! :+1:

tafaRU commented on account_fiscal_year/__openerp__.py:
```diff
@@ -0,0 +1,30 @@
+        'data/date_range_type.xml',
+        'views/date_range_type.xml',
+    ],
+    'test': [
```
Please remove the following three empty keys.

tafaRU commented:
A little nitpicking, but :+1:

Thanks!

damdam-s committed:
[FIX] clean __openerp__.py

lepistone commented:
Thanks! :+1:

lmignon commented:
:+1: Thanks!
